quill-cassandra-ce/src/main/scala-2.12/io/getquill/util/GuavaCeUtils.scala

```diff
@@ -0,0 +1,47 @@
+package io.getquill.util
+
+import com.google.common.util.concurrent.{ FutureCallback, Futures, ListenableFuture }
+import cats.effect.Async
+
+import scala.concurrent.ExecutionContext.global
+import cats.implicits._
+import com.datastax.driver.core.{ ResultSet, ResultSetFuture }
+
+import java.util.concurrent.Executor
+import scala.language.higherKinds
+
+object GuavaCeUtils {
+
+  case object ecExecutor extends Executor {
+    def execute(command: Runnable): Unit = global.execute(command)
+  }
+
+  implicit class RichResultSetFuture[F[_]: Async](rsf: F[ResultSetFuture]) {
+    def toAsync: F[ResultSet] = {
+      rsf.flatMap { lf =>
+        Async[F].async_ { cb =>
+          Futures.addCallback(lf, new FutureCallback[ResultSet] {
+            override def onFailure(t: Throwable): Unit = cb(Left(t))
+
+            override def onSuccess(result: ResultSet): Unit = cb(Right(result))
+          }, ecExecutor)
```

Comment:
A better approach here would be to get the `Async[F].executionContext`, then wrap it into an `Executor` by delegating `execute`. All uses of `global` are highly suspect.

Reply:
This makes sense, @adamgfraser did the same thing in `FutureConversions.scala` in `ListenableFutureConverter.asScala` to help with ZIO interop. 
Maybe something like this?
```scala
val asyncEcExecutor =
  new Executor {
    override def execute(command: Runnable): Unit = Async[F].executionContext.execute(command)
  }
```